### PR TITLE
:book: Add cluster-resource-set-controller to docs

### LINKED ISSUE #5475: Add documentation for controllers missing from book
**User Story**

As a user/operator I would like comprehensive documentation of all CAPI controllers.

**Detailed Description**

Currently there is documentation in section 3.5 of the CAPI book for most of the controllers in CAPI but some are missing. 

Documentation could be added for the following controllers:

**Core controllers**
- [x]  Topology/Cluster Controller #5502
- [ ] ClusterResourceSet/ClusterResourceSetBinding

**Internal controllers**
- [ ] ClusterCache
- [ ] Topology/MachineDeployment
- [ ] Topology/MachineSet

c/f #5471 

/kind feature
/kind documentation

### PR DESCRIPTION
**Pull request by ramineni**
**What this PR does / why we need it**:
Add a documentation page on ClusterResourceSet controller
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5475

### CODE REVIEW

File: docs/book/src/developer/architecture/controllers/cluster-resource-set.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# `ClusterResourceSet` Controller
+
+The `ClusterResourceSet` provide a mechanism for applying resources in a cluster once it is created.
+
+![ClusterResourceSet Controller Activity Diagram](../../../images/cluster-resource-set-controller.png)
+
+Its main responsibilities are to:
+1. Provide a means to specify a set of resources to apply automatically to newly-created and existing Clusters. Resources will be applied only once.
```
Comment (killianmuldoon) [inline docs/book/src/developer/architecture/controllers/cluster-resource-set.md:8]:
```suggestion
Its main responsibility is to automatically apply a set of resources to newly-created and existing Clusters. Resources will be applied only once.
```

File: docs/book/src/developer/architecture/controllers/cluster-resource-set.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# `ClusterResourceSet` Controller
+
+The `ClusterResourceSet` provide a mechanism for applying resources in a cluster once it is created.
```
Comment (killianmuldoon) [inline docs/book/src/developer/architecture/controllers/cluster-resource-set.md:3]:
```suggestion
The `ClusterResourceSet` provides a mechanism for applying resources in a cluster once it is created.
```

File: docs/book/src/developer/architecture/controllers/cluster-resource-set.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+![ClusterResourceSet Controller Activity Diagram](../../../images/cluster-resource-set-controller.png)
+
+Its main responsibilities are to:
+1. Provide a means to specify a set of resources to apply automatically to newly-created and existing Clusters. Resources will be applied only once.
+2. Additions to the resource list to be reconciled immediately and will be applied to both new and existing matching clusters.
```
Comment (killianmuldoon) [inline docs/book/src/developer/architecture/controllers/cluster-resource-set.md:9]:
```suggestion
```

File: docs/book/src/developer/architecture/controllers/cluster-resource-set.md (hunk 2 above)
Comment (killianmuldoon) [inline docs/book/src/developer/architecture/controllers/cluster-resource-set.md:3]:
```suggestion
The `ClusterResourceSet` provide a mechanism for applying resources - e.g. pods, deployments, daemonsets - to a cluster once it is created.
```

Comment (ramineni) [reply docs/book/src/developer/architecture/controllers/cluster-resource-set.md:3]: @killianmuldoon little confused here, resource types should be of Secrets/ConfigMaps right?

Comment (ramineni) [reply docs/book/src/developer/architecture/controllers/cluster-resource-set.md:3]: got it. Thanks.

**Review by killianmuldoon — COMMENTED** [review]
One last nit  - could you move the position in Summary.md to Just below the ClusterTopology controller?

File: docs/book/src/SUMMARY.md [hunk 4]
```diff
@@ -62,6 +62,7 @@
         - [Metadata propagation](./developer/architecture/controllers/metadata-propagation.md)
```
Comment (killianmuldoon) [inline docs/book/src/SUMMARY.md:62]: This is where I mean - move the ClusterResourceSet up above Metadata propagation and below ClusterTopology in the summary

Comment (ramineni) [reply docs/book/src/SUMMARY.md:63]: Done

File: docs/book/src/images/cluster-resource-set-controller.plantuml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      endif
+    :Get Resoures in clusterResourceSet;
+    if (Check applied condition of resource in resourceSetBinding) then (false)
+     :Apply resource to cluster;
+     :Set applied condition to true in resourceSetBinding;
+    endif
+    :Mark ResourcesAppliedCondition to true;
+  endif
+  repeat while (More clusters?)
+repeat while (More updates to clusterResourceSet?)
```
Comment (sbueringer) [inline docs/book/src/images/cluster-resource-set-controller.plantuml:26]:
```suggestion
    if (Cluster matches ClusterResourceSet's selector) then (yes)
      if (Get ClusterResourceSetBinding for the cluster) then (exists)
      else (doesn't exist)
         :Create ClusterResourceSetBinding object for the cluster;
      endif
      if (ResourceSetBinding for a given ClusterResourceSet exists) then (yes)
      else (no)
        :Create ResourceSetBinding;
      endif
    :Get Resoures in ClusterResourceSet;
    if (Check applied condition of resource in ResourceSetBinding) then (false)
     :Apply resource to cluster;
     :Set applied condition to true in ResourceSetBinding;
    endif
    :Mark ResourcesAppliedCondition to true;
  endif
  repeat while (More clusters?)
repeat while (More updates to ClusterResourceSet?)
```
nit: let's use upper case when we refer to literal API types

**Comment by sbueringer** [comment]
Thx!

Please squash the commits, otherwise lgtm from my side

**Pull request merged**